### PR TITLE
Multiplacements

### PR DESCRIPTION
**Pull request by wasade**
Handle fragments with multiple placement entries

cc @antgonza

### CODE REVIEW

File: bp/tests/test_io.py [hunk 1]
```diff
@@ -215,6 +215,50 @@ def test_parse_jplace_simple(self):
...
+        for placement in data['placements']:
+            if placement['n'][0] == '82':
+                keep.append(placement)
+            elif placement['n'][0] == '99':
+                placement['p'].append([309, 0.04520741687623886, 1,
+                                       0.11020044356641526,
+                                       0.00550337922097477])
+                keep.append(placement)
+            elif placement['n'][0] == '55':
+                print(placement['p'])
```
Comment (antgonza) [inline bp/tests/test_io.py:247]: rm prints?

File: bp/tests/test_io.py [hunk 2]
```diff
@@ -215,6 +215,50 @@ def test_parse_jplace_simple(self):
...
+                print("***")
+                keep.append(placement)
+        data['placements'] = keep
+        data = json.dumps(data)
+        exp_tree = self.tree
+        obs_df, obs_tree = parse_jplace(data)
+        obs_tree = to_skbio_treenode(obs_tree)
+        pdt.assert_frame_equal(obs_df, exp_df)
+        self.assertEqual(obs_tree.compare_rfd(exp_tree), 0)
```
Comment (antgonza) [inline bp/tests/test_io.py:261]: Just wondering why not just do:
```self.assertEqual(obs_tree.compare_rfd(self.tree), 0)```
and remove 257? flake8?

**Review by antgonza — COMMENTED** [review]
Looks good; should I test this version with the SEPP/qiita fragments before approving?

File: bp/tests/test_io.py (hunk 1 above)
Comment (wasade) [reply bp/tests/test_io.py:247]: good catch thanks

File: bp/tests/test_io.py (hunk 2 above)
Comment (wasade) [reply bp/tests/test_io.py:261]: just so the variable name is explicit

**Comment by wasade** [comment]
re: qiita, ya that would be good -- our expectation is << memory and walltime

**Comment by antgonza** [comment]
K, note that I forgot to mention this in the previous PR but it would be good to add:
```
bp/GPL/_*.c
bp/_*.c
bp/tests/test_bp_cy.c
```
to the gitignore ...

**Comment by wasade** [comment]
The advice on that w/ cython has changed over the years, but can do

**Comment by antgonza** [comment]
90 bp tree previous version:
```
01/20/2022 13:56:44  S    Exit_status=0 resources_used.cput=01:57:31 resources_used.mem=108504636kb resources_used.vmem=109014944kb resources_used.walltime=01:58:52
```

this PR:
```
01/24/2022 16:10:13  S    Exit_status=0 resources_used.cput=00:22:49 resources_used.mem=40521136kb resources_used.vmem=43169648kb resources_used.walltime=00:23:45
```

Note that the resulting tree of this PR is ~1/6 of the previous run; is that expected? 

```bash
$ ls test1 test1-files/test1 -lah
-rw-rw-r-- 1 qiita_t qiita 1.1G Jan 24 16:09 test1
-rw-rw-r-- 1 qiita_t qiita 6.7G Jan 20 13:50 test1-files/test1
```

**Comment by antgonza** [comment]
Just to leave it here, for 150 bps; original:
```
01/21/2022 11:34:15  S    Exit_status=0 resources_used.cput=03:31:01 resources_used.mem=207398712kb resources_used.vmem=207909100kb resources_used.walltime=03:32:03
```
this PR: 
```
01/25/2022 07:00:05  S    Exit_status=0 resources_used.cput=00:47:58 resources_used.mem=83402824kb resources_used.vmem=85256120kb resources_used.walltime=00:48:46
```

Same with size:
```
$ ls -lah test2 test2-files/test2
-rw-rw-r-- 1 qiita_t qiita 3.5G Jan 25 06:58 test2
-rw-rw-r-- 1 qiita_t qiita  20G Jan 21 11:24 test2-files/test2
```

**Comment by wasade** [comment]
Thanks!! I think these look good and fit general expectations. Can you
share the path for the 90nt?

**Comment by antgonza** [comment]
/panfs/panfs1.ucsd.edu/panscratch/qiita_t/working_directory/placement-benchmark/
bp-90bp.tre
bp-150bp.tre

**Comment by wasade** [comment]
Thanks!

**Comment by wasade** [comment]
@antgonza are we good on this given our sepp/sepp tests yesterday?

**Comment by wasade** [comment]
Thanks!

I'm working on moving this over to biocore, at which point permissions will be addressed

**Pull request merged**